models/nft/fantom/nft_fantom_transfers.sql

```diff
@@ -0,0 +1,126 @@
+     , 'single'                             AS transfer_type
+     , t.evt_index
+     , t.contract_address
+     , t.id                                 AS token_id
+     , t.value                              AS amount
+     , t.from
+     , t.to
+     , ft.from as executed_by
+     , t.evt_tx_hash                        AS tx_hash
+     , 'fantom' || t.evt_tx_hash || '-erc721-' || t.contract_address || '-' || t.id || '-' || t.from || '-' || t.to ||
+       '-' || t.value || '-' || t.evt_index AS unique_transfer_id
+FROM {{ source('erc1155_fantom', 'evt_transfersingle') }} t
+INNER JOIN {{ source('fantom', 'transactions') }} ft 
+    ON ft.block_number = t.evt_block_number
+    AND ft.hash = t.evt_tx_hash
+{% if is_incremental() %}
+    AND ft.block_time >= date_trunc("day", now() - interval '1 week')
+ANTI JOIN {{this}} anti_table
+    ON t.evt_tx_hash = anti_table.tx_hash
+    AND anti_table.block_time >= date_trunc("day", now() - interval '1 week')
+WHERE
+    t.evt_block_time >= date_trunc("day", now() - interval '1 week')
+{% endif %}
+
+UNION ALL
+
+SELECT 'fantom'                          AS blockchain
+     , t.evt_block_time                    AS block_time
+     , date_trunc('day', t.evt_block_time) AS block_date
+     , t.evt_block_number                  AS block_number
+     , 'erc1155'                         AS token_standard
+     , 'batch'                           AS transfer_type
+     , t.evt_index
+     , t.contract_address
+     , t.ids_and_count.ids                 AS token_id
+     , t.ids_and_count.values              AS amount
+     , t.from
+     , t.to
+     , ft.from as executed_by
+     , t.evt_tx_hash AS tx_hash
+     , 'fantom' || t.evt_tx_hash || '-erc1155-' || t.contract_address || '-' || t.ids_and_count.ids || '-' || t.from || '-' ||
+        t.to || '-' || t.ids_and_count.values || '-' || t.evt_index AS unique_transfer_id
+FROM
+(
+    SELECT t.evt_block_time
+           , t.evt_block_number
+           , t.evt_tx_hash
+           , t.contract_address
+           , t.from
+           , t.to
+           , t.evt_index
+           , explode(arrays_zip(t.values, t.ids)) AS ids_and_count
+    FROM {{ source('erc1155_fantom', 'evt_transferbatch') }} t
+        {% if is_incremental() %}
+        ANTI JOIN {{this}} anti_table
+            ON t.evt_tx_hash = anti_table.tx_hash
+            AND anti_table.block_time >= date_trunc("day", now() - interval '1 week')
+        WHERE t.evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+    GROUP BY t.evt_block_time, t.evt_block_number, t.evt_tx_hash, t.contract_address, t.from, t.to, t.evt_index, t.values, t.ids
+) t 
+INNER JOIN {{ source('fantom', 'transactions') }} ft 
+    ON ft.block_number = t.evt_block_number
+    AND ft.hash = t.evt_tx_hash
+    {% if is_incremental() %}
+    AND ft.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+WHERE ids_and_count.values > 0
+GROUP BY 
```

Comment:
what is the intention of the group by here, if we aren't aggregating any columns? is it solely for unique rows returned? if so, i'd prefer to see fixes to the logic to ensure unique rows -- is that possible?

Reply:
hmm, lemme go over the code again @jeff-dude, I originally just copy-pasted what was in the other nft transfer models and updated the tables.

Reply:
fair, i see that in other models. i won't question it for now. it'd be a universal change if anything.